File: StudentList.py
def displayStudent(map: dict): 
    print('\nStudentList')
    for [key, value] in map.items():
        print(f'{key} {value}')
        
        
def changeLastData(map: dict, lastElement): 
    studNumber = input('Enter your student Number : ')
    studname = input('Enter your studentName : ')
    
    #removing the lastelement
    del map[lastElement]
    #adding the new element 
    map[studNumber] = studname
    
    
def studentList():
    students = {}
    lastElement = ""
    
    for i in range(3): 
        studNumber = input("Enter your student Number : ")
        studName = input("Enter your first name : ")
        
        #adding the key value pairs 
        students[studNumber] = studName
        #adding the last element to this variable for modification
        lastElement = studNumber
    
    #displaying the current element
    displayStudent(students)
    #changing the last element
    changeLastData(students, lastElement)
    #displaying it again
    displayStudent(students)
    
studentList()

File: StudentList.java
import java.util.Scanner;
import java.util.HashMap;
import java.util.Map;

public class StudentList {
    static Map<String, String> students = new HashMap<>();
    static Scanner scanner = new Scanner(System.in);
    static String lastKey = ""; //to store the last key

    public static void main(String[] args) {

        for (int i = 0; i < 3; i++) {
            System.out.print("Enter your student number : ");
            String studNumber = scanner.nextLine();
            System.out.print("Enter your first name : ");
            String firstName = scanner.nextLine();

            students.put(studNumber, firstName);
            lastKey = studNumber;// will be used to remove the last element 
        }
        DisplayVal(students);
        changeData();//overriding the value of the prev entries
        DisplayVal(students);

    }
    static void DisplayVal(Map<String, String>  n){
        System.out.println("\nStudent List");
        for(Map.Entry<String, String> e : students.entrySet()){
            System.out.printf("%s %s\n", e.getKey(), e.getValue());
        }
    }
    static void changeData(){
        System.out.print("Enter your student number : ");
        String studNum = scanner.nextLine();
        System.out.print("Enter your firstname : ");
        String firstName = scanner.nextLine();

        students.remove(lastKey);//remove the last prev elment
        students.put(studNum, firstName);//add the new element
    }
}

